src/core/product-transformer.ts

```diff
@@ -52,6 +52,20 @@ namespace Utils {
     }
   }
 
+  export function combineData(data: object, variant: object) {
+    const newData = {...data};
+
+    // overwrite only if not undefined
+    Object.keys(variant).forEach((key) => {
```

Comment:
Consider [`Array.prototype.reduce()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)

Reply:
Replace it with this? Not sure if it's a good thing for readability
```js
    // overwrite only if not undefined
    return Object.keys(variant).reduce((newData, key) => {
      const value = variant[key];
      if (value !== undefined) {
        newData[key] = value;
      }
      return newData;
    }, {...data});
```